adserver/tasks.py

```diff
@@ -281,3 +284,21 @@ def remove_old_client_ids(days=90):
         offers_changed = Offer.objects.filter(pk__in=offer_ids).update(client_id=None)
         if not offers_changed:
             break
+
+
+@app.task()
+def calculate_publisher_ctrs(days=7):
+    """Calculate average CTRs for paid ads on a publisher for the last X days."""
+    sample_cutoff = get_ad_day() - datetime.timedelta(days=days)
+
+    for publisher in Publisher.objects.all():
+        queryset = AdImpression.objects.filter(
+            date__gte=sample_cutoff,
+            publisher=publisher,
+            advertisement__flight__campaign__campaign_type=PAID_CAMPAIGN,
+        )
+        report = PublisherReport(queryset)
+        report.generate()
```

Comment:
There's probably a more efficient way to do this with a query, but this is fine and incentivizes us to make report generation as efficient as possible 👍